modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DotExpanderProcessor.java

```diff
@@ -64,8 +64,8 @@ public IngestDocument execute(IngestDocument ingestDocument) throws Exception {
 
     private void expandDot(IngestDocument ingestDocument, String pathToExpand, String fieldName, Map<String, Object> map) {
         if (map.containsKey(fieldName)) {
+            Object value = map.remove(fieldName);
```

Comment:
I moved this out of the condition as `hasField` can now resolve dotted fields. So if we want to expand the `fieldName` `"foo.bar"`, and the doc contains both `['foo']['bar']` and `['foo.bar']`, `hasField("foo.bar")` will always be `true`